gui/wxpython/mapdisp/settings.py

```diff
@@ -0,0 +1,276 @@
+"""
+@package mapdisp.settings
+
+@brief Classes for map display settings management
+
+Classes:
+ - settings::ChBItem
+ - settings::ChBRender
+ - settings::ChBShowRegion
+ - settings::ChBAlignExtent
+ - settings::ChBResolution
+ - settings::MapDisplayPreferencesDialog
+
+(C) 2021 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author Vaclav Petras <wenzeslaus gmail.com>
+@author Anna Kratochvilova <kratochanna gmail.com>
+@author Linda Kladivova <lindakladivova gmail.com>
+"""
+
+import wx
+import wx.lib.scrolledpanel as SP
+from gui_core.wrap import StaticBox
+
+from core.settings import UserSettings
+from gui_core.preferences import PreferencesBaseDialog
+
+
+class ChBItem(wx.CheckBox):
+    """Base class for Map Display settings checkboxes that use property signals"""
+
+    def __init__(self, parent, mapWindowProperties, label, tooltip):
+        wx.CheckBox.__init__(
+            self, parent=parent, id=wx.ID_ANY, label=label, name="IsChecked"
+        )
+        self._properties = mapWindowProperties
+        self._setValue(self.mapWindowProperty)
+        self.SetToolTip(wx.ToolTip(tooltip))
+
+        self._connect()
+        self.Bind(wx.EVT_CHECKBOX, self._onToggleCheckBox)
+
+    @property
+    def mapWindowProperty(self):
+        pass
+
+    @mapWindowProperty.setter
+    def mapWindowProperty(self, value):
+        pass
+
+    def mapWindowPropertyChanged(self):
+        """Returns signal from MapWindowProperties."""
+        pass
+
+    def _setValue(self, value):
+        self.SetValue(value)
+
+    def SetValue(self, value):
+        self._disconnect()
+        self.mapWindowProperty = value
+        wx.CheckBox.SetValue(self, value)
+        self._connect()

```

Comment:
I don't see the need for both `_setValue` and `SetValue`.

Shouldn't SetValue just set the checkbox, not the property? Setting the property would be done in `_onToggleCheckBox`.